cloudigrade/util/tests/test_aws.py

```diff
@@ -12,25 +12,84 @@
 from django.test import TestCase
 
 from util import aws
-from util.aws import extract_account_id_from_arn
+from util.aws import AwsArn
 from util.tests import helper
 
 
 class UtilAwsTest(TestCase):
     """AWS utility functions test case."""
 
-    def test_extract_account_id_from_arn(self):
-        """Assert successful account ID extraction from a well-formed ARN."""
+    def test_parse_arn_with_region_and_account(self):
+        """Assert successful account ID parsing from a well-formed ARN."""
         mock_account_id = helper.generate_dummy_aws_account_id()
-        mock_arn = helper.generate_dummy_arn(mock_account_id)
-        extracted_account_id = aws.extract_account_id_from_arn(mock_arn)
-        self.assertEqual(mock_account_id, extracted_account_id)
+        mock_arn = helper.generate_dummy_arn(account_id=mock_account_id,
+                                             region='test-region-1')
 
-    def test_error_extract_account_id_from_invalid_arn(self):
-        """Assert error in account ID extraction from a badly-formed ARN."""
+        arn_object = AwsArn(mock_arn)
+
+        partition = arn_object.partition
+        self.assertNotEqual(partition, None)
```

Comment:
You can also use `assertIsNone` or `assertIsNotNone`

Reply:
To Andrew's point, yeah, let's change these to identity checks against singletons like `None` instead of equality checks before moving forward. Technically equality checks _may_ work, but sometimes they don't (since you can override the eq operator behavior) and the prevailing standards and conventions say we shouldn't rely on them.

https://www.python.org/dev/peps/pep-0008/#programming-recommendations

> Comparisons to singletons like None should always be done with is or is not, never the equality operators.